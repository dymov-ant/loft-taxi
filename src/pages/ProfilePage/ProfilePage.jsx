import React, { Component } from "react"
import Card from "../../components/Card"
import ProfileForm from "../../components/ProfileForm"
import style from "./profilePage.module.scss"

class ProfilePage extends Component {
  render() {
    return (
      <div className={style.profile}>
        <Card>
          <ProfileForm/>
        </Card>
      </div>
    )
  }
}

export default ProfilePage