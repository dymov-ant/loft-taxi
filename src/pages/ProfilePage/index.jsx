import React, { PureComponent } from 'react';
import Card from '../../components/Card';
import ProfileForm from '../../components/ProfileForm';
import style from './profilePage.module.scss';

class ProfilePage extends PureComponent {
  render() {
    return (
      <div className={style.profile}>
        <div className={style.profile__card}>
          <Card>
            <ProfileForm />
          </Card>
        </div>
      </div>
    );
  }
}

export default ProfilePage;
