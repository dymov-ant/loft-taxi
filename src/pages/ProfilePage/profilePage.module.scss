.profile {
  display: flex;
  justify-content: center;
  align-items: center;
  height: calc(100vh - 106px);
  background: linear-gradient(0deg, rgba(28, 26, 25, 0.5), rgba(28, 26, 25, 0.5)), url("../../assets/img/map-bg.png") no-repeat center / cover;
}