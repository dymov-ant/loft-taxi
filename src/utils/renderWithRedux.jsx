import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { createMemoryHistory } from 'history';
// eslint-disable-next-line import/no-extraneous-dependencies
import configureStore from 'redux-mock-store';

const history = createMemoryHistory();
const mockStore = configureStore();

export const renderWithRedux = (component, initialState = {}) => {
  const store = mockStore(initialState);
  return render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  );
};
