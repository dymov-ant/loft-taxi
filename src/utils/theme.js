import { createTheme } from "@material-ui/core"
import { orange } from "@material-ui/core/colors"

export const theme = createTheme({
  palette: {
    primary: {
      main: "#ffc617"
    },
    secondary: {
      main: "#ff0000"
    }
  },
  status: {
    danger: orange[500]
  },
  overrides: {
    MuiFormLabel: {
      root: {
        "&$focused": {
          color: "#1473e6"
        }
      }
    },
    MuiLink: {
      root: {
        color: "#1473e6",
        cursor: "pointer"
      }
    },
    MuiInput: {
      underline: {
        "&:after": {
          "border-bottom-color": "#1473e6"
        }
      }
    },
    MuiFormHelperText: {
      root: {
        position: "absolute",
        bottom: "-1.4em"
      }
    },
    MuiButton: {
      root: {
        textTransform: "none"
      },
      containedPrimary: {
        boxShadow: "none",
        backgroundColor: "#ffc617",
        "&:focus": {
          boxShadow: "0 0 0 0.2rem rgba(0,123,255,.5)"
        }
      }
    }
  }
})