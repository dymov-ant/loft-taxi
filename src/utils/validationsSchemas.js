import * as yup from "yup"


export const loginSchema = yup.object({
  email: yup
    .string()
    .email("Введите корректный email")
    .required("Поле обязательно"),
  password: yup
    .string()
    .min(6, "Минимум 6 символов")
    .required("Поле обязательно")
})

export const registrationSchema = yup.object({
  email: yup
    .string()
    .email("Введите корректный email")
    .required("Поле обязательно"),
  name: yup
    .string()
    .required("Поле обязательно"),
  surname: yup
    .string()
    .required("Поле обязательно"),
  password: yup
    .string()
    .min(6, "Минимум 6 символов")
    .required("Поле обязательно")
})