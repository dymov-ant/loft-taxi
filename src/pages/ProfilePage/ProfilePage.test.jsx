import React from 'react';
import { renderWithRedux } from '../../utils/renderWithRedux';
import ProfilePage from './index';

describe('Profile page', () => {
  const initialState = {
    profile: {
      isLoading: false,
    },
  };
  it('ProfilePage renders', () => {
    const { container } = renderWithRedux(<ProfilePage />, initialState);

    expect(container).toBeInTheDocument();
  });

  it('MapPage snapshot', () => {
    const profilePage = renderWithRedux(<ProfilePage />, initialState);

    expect(profilePage).toMatchSnapshot();
  });
});
