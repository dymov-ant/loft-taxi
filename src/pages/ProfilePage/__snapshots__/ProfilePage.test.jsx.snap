// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page MapPage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile"
      >
        <div
          class="profile__card"
        >
          <div
            class="card"
          >
            <div
              class="form__wrapper"
            >
              <h1
                class="form__title"
              >
                Профиль
              </h1>
              <p
                class="form__text"
              >
                Введите платежные данные
              </p>
              <form
                class="form__formWrapper"
                novalidate=""
              >
                <div
                  class="form__form"
                >
                  <div
                    class="form__row"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                        data-shrink="false"
                      >
                        Имя владельца
                      </label>
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          name="cardName"
                          style="text-transform: uppercase;"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="form__row"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                        data-shrink="false"
                      >
                        Номер карты
                      </label>
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          name="cardNumber"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="form__row"
                  >
                    <div
                      class="form__col"
                    >
                      <div
                        class="MuiFormControl-root MuiTextField-root makeStyles-input-2"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                          data-shrink="true"
                        >
                          MM/YY
                        </label>
                        <div
                          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                            name="expiryDate"
                            type="text"
                            value="10/21"
                          />
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                          >
                            <button
                              class="MuiButtonBase-root MuiIconButton-root"
                              tabindex="0"
                              type="button"
                            >
                              <span
                                class="MuiIconButton-label"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                                  />
                                  <path
                                    d="M0 0h24v24H0z"
                                    fill="none"
                                  />
                                </svg>
                              </span>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="form__col"
                    >
                      <div
                        class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                          data-shrink="false"
                        >
                          CVC
                        </label>
                        <div
                          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiInput-input"
                            name="cvc"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="form__cardWrapper"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
                  >
                    Нет сохраненных карт
                  </h6>
                </div>
              </form>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                data-testid="button"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  Сохранить
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile"
    >
      <div
        class="profile__card"
      >
        <div
          class="card"
        >
          <div
            class="form__wrapper"
          >
            <h1
              class="form__title"
            >
              Профиль
            </h1>
            <p
              class="form__text"
            >
              Введите платежные данные
            </p>
            <form
              class="form__formWrapper"
              novalidate=""
            >
              <div
                class="form__form"
              >
                <div
                  class="form__row"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink="false"
                    >
                      Имя владельца
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        name="cardName"
                        style="text-transform: uppercase;"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="form__row"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink="false"
                    >
                      Номер карты
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        name="cardNumber"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="form__row"
                >
                  <div
                    class="form__col"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root makeStyles-input-2"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                        data-shrink="true"
                      >
                        MM/YY
                      </label>
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                          name="expiryDate"
                          type="text"
                          value="10/21"
                        />
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                        >
                          <button
                            class="MuiButtonBase-root MuiIconButton-root"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                                />
                                <path
                                  d="M0 0h24v24H0z"
                                  fill="none"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="form__col"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                        data-shrink="false"
                      >
                        CVC
                      </label>
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          name="cvc"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form__cardWrapper"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
                >
                  Нет сохраненных карт
                </h6>
              </div>
            </form>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="button"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Сохранить
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
